kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/KubernetesAssistant.java

```diff
@@ -0,0 +1,482 @@
+
+import io.fabric8.kubernetes.api.model.v3_1.EndpointSubset;
+import io.fabric8.kubernetes.api.model.v3_1.Endpoints;
+import io.fabric8.kubernetes.api.model.v3_1.HasMetadata;
+import io.fabric8.kubernetes.api.model.v3_1.Pod;
+import io.fabric8.kubernetes.api.model.v3_1.ReplicationController;
+import io.fabric8.kubernetes.api.model.v3_1.Service;
+import io.fabric8.kubernetes.api.model.v3_1.ServicePort;
+import io.fabric8.kubernetes.api.model.v3_1.extensions.Deployment;
+import io.fabric8.kubernetes.clnt.v3_1.ConfigBuilder;
+import io.fabric8.kubernetes.clnt.v3_1.KubernetesClient;
+import io.fabric8.kubernetes.clnt.v3_1.KubernetesClientException;
+import io.fabric8.kubernetes.clnt.v3_1.dsl.NamespaceListVisitFromServerGetDeleteRecreateWaitApplicable;
+import io.fabric8.kubernetes.clnt.v3_1.internal.readiness.Readiness;
+import io.github.lukehutch.fastclasspathscanner.FastClasspathScanner;
+import io.github.lukehutch.fastclasspathscanner.matchprocessor.FileMatchProcessor;
+import org.arquillian.cube.kubernetes.impl.portforward.PortForwarder;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.ServerSocket;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.arquillian.cube.kubernetes.impl.enricher.KuberntesServiceUrlResourceProvider.LOCALHOST;
+import static org.awaitility.Awaitility.await;
+
+/**
+ * Class that allows you to deploy undeploy and wait for resources programmatically in a test.
+ */
+public class KubernetesAssistant {
+
+    private static final Logger log = Logger.getLogger(KubernetesAssistant.class.getName());
+    private final io.fabric8.kubernetes.clnt.v3_1.KubernetesClient client;
+
+    private final String namespace;
+    private String applicationName;
+
+    private KubernetesAssistantDefaultResourceLocator kubernetesAssistantDefaultResourcesLocator;
+
+    private Map<String, List<HasMetadata>> created = new LinkedHashMap<>();
+
+    public KubernetesAssistant(io.fabric8.kubernetes.clnt.v3_1.KubernetesClient kubernetesClient, String namespace) {
+        this.client = kubernetesClient;
+        this.namespace = namespace;
+        this.kubernetesAssistantDefaultResourcesLocator = new KubernetesAssistantDefaultResourceLocator();
+    }
+
+    /**
+     * Deploys application finding resources in default location in classpath. That is:
+     * openshift.(y[a]ml|json), kubernetes.(y[a]ml|json), META-INF/fabric8/openshift.(y[a]ml|json), META-INF/fabric8/kubernetes.(y[a]ml|json)
```

Comment:
Update javadoc because my idea is that in case of using kubernetes, `openshift` resource should not be loaded.